src/main/java/io/zksync/signer/EthSigner.java

```diff
@@ -1,120 +1,102 @@
-                MnemonicUtils.generateSeed(mnemonic, ""));
+import io.zksync.domain.token.Token;
 
-        // Derive the keypair using the derivation path
-        Bip32ECKeyPair  derivedKeyPair = Bip32ECKeyPair.deriveKeyPair(masterKeypair, derivationPath);
+public interface EthSigner {
+    
+    /**
+     * Get wallet address
+     * 
+     * @return Address in hex string
+     */
+    String getAddress();
+
+    /**
+     * Get internal credentials. @see org.web3j.crypto.Credentials
+     * 
+     * @return Credentials object that contains private and public key pair
+     */
+    Credentials getCredentials();
+
+    /**
+     * Sign ZkSync `ChangePubKey` operation message
+     * 
+     * @param pubKeyHash - ZkSync public key hash
+     * @param nonce - Nonce of the account
+     * @param accountId - Id of the account in ZkSync
+     * @return Signature object
+     */
+    EthSignature signChangePubKey(String pubKeyHash, Integer nonce, Integer accountId);
+
+    /**
+     * Sign ZkSync `Transfer` operation message
+     * 
+     * @param to - Address of receiver
+     * @param accountId - Id of the sender account in ZkSync
+     * @param nonce - Nonce of the sender account
+     * @param amount - Amount tokens to transfer
+     * @param token - Token object supported by ZkSync
+     * @param fee - Cost of transaction in ZkSync network
+     * @return Signature object
+     */
+    EthSignature signTransfer(String to, Integer accountId, Integer nonce, BigInteger amount, Token token, BigInteger fee);
+
+    /**
+     * Sign ZkSync `Withdraw` operation message
+     * 
+     * @param to - Address of receiver
+     * @param accountId - Id of the account in ZkSync
+     * @param nonce - Nonce of the account
+     * @param amount - Amount tokens to withdraw
+     * @param token - Token object supported by ZkSync
+     * @param fee - Cost of transaction in ZkSync network
+     * @return Signature object
+     */
+    EthSignature signWithdraw(String to, Integer accountId, Integer nonce, BigInteger amount, Token token, BigInteger fee);
+
+    /**
+     * Sign raw message
+     * 
+     * @param message - Message to sign
+     * @return Signature object
+     */
+    EthSignature signMessage(String message);
+
+    /**
+     * Sign raw message
+     * 
+     * @param message - Message to sign
+     * @param addPrefix - If true then add secure prefix (https://eips.ethereum.org/EIPS/eip-712)
+     * @return
+     */
+    EthSignature signMessage(String message, boolean addPrefix);
+
+    /**
+     * Verify signature with raw message
+     * 
+     * @param signature - Signature object
+     * @param message - Message to verify
+     * @return true on verification success
+     * @throws SignatureException If the public key could not be recovered or if there was a
+     *     signature format error.
+     */
+    boolean verifySignature(EthSignature signature, String message) throws SignatureException;
```

Comment:
Verification functions should be async too.

There are EIP-1271 signatures, verification of which requires an async call.